internal/graph/graph.go

```diff
@@ -238,10 +302,28 @@ func (g *ConnectedObjectGraph) findIngressesWithTargetRewrite(
 			//     define parent: [folder]
 			//     define viewer: viewer from parent
 			if typeRestriction.GetType() == source.GetType() && computedUserset == source.GetRelation() {
+
+				condition := NoFurtherEvalCondition
+
+				involvesIntersection, err := g.typesystem.RelationInvolvesIntersection(typeRestriction.GetType(), r.GetName())
+				if err != nil {
+					return nil, err
+				}
+
+				involvesExclusion, err := g.typesystem.RelationInvolvesExclusion(typeRestriction.GetType(), r.GetName())
```

Comment:
Wouldn't it be more optimal if this was: `ts.RelationInvolvesIntersectionOrExclusion`? That way you don't have to traverse twice to compute